plugins/basic/nexus-repository-maven/src/main/java/org/sonatype/nexus/repository/maven/internal/maven2/Maven2MavenPathParser.java

```diff
@@ -106,23 +108,41 @@ else if (str.endsWith("." + HashType.MD5.getExt())) {
           tail = str.substring(nTailPos);
         }
         else {
-          StringBuilder snapshotBuildNumber = new StringBuilder(version);
-          snapshotBuildNumber
-              .append(str.substring(vSnapshotStart + version.length(), vSnapshotStart + version.length() + 8));
+          final StringBuilder snapshotTimestampedVersion = new StringBuilder(version);
+          snapshotTimestampedVersion.append(
+              str.substring(vSnapshotStart + version.length(), vSnapshotStart + version.length() + 7)
```

Comment:
is the change from 8 to 7 intentional? there's a lot of magic numbers in this code, I wonder if it makes more sense to use a pattern matcher or Guava splitter? otherwise introducing some constants would make this code more readable.